org.lflang/src/org/lflang/validation/ValidatorErrorReporter.java

```diff
@@ -0,0 +1,158 @@
+
+import java.nio.file.Path;
+
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.xtext.validation.ValidationMessageAcceptor;
+import org.lflang.ErrorReporter;
+
+/**
+ * This class translates messages reported via the ErrorReporrter interface to
+ * the interface of a given ValidationMessageAcceptor.
+ * 
+ * Effectively this allows to report errors via the ErrorReporter interface
+ * during validator checks, while having the validator still track all the
+ * reported warnings and messages. This is required for some functionality, like
+ * the construction of an instance graph in LFValidator.checkModel(). Since the
+ * instance graph is also used in other components, it does not report directly
+ * to the validator, but uses our custom ErrorReporter interface that we use
+ * during code generation. This class bridges the gap between the ErrorReporter
+ * interface and the messages that the validator expects.
+ * 
+ * @author Christian Menard <christian.menard@tu-dresden.de>
+ */
+public class ValidatorErrorReporter implements ErrorReporter {
+
+    private ValidationMessageAcceptor acceptor;
+    private BaseLFValidator.ValidatorStateAccess validatorState;
+    private boolean errorsOccurred = false;
+
+    public ValidatorErrorReporter(ValidationMessageAcceptor acceptor,
+            BaseLFValidator.ValidatorStateAccess stateAccess) {
+        this.acceptor = acceptor;
+        this.validatorState = stateAccess;
+    }
+
+    /**
+     * Report the given message as an error on the object currently under
+     * validation.
+     */
+    @Override
+    public String reportError(String message) {
+        errorsOccurred = true;
+        acceptor.acceptError(message, validatorState.getCurrentObject(), null,
+                ValidationMessageAcceptor.INSIGNIFICANT_INDEX, null);
+        return message;
+    }
+
+    /**
+     * Report the given message as an error on the given object.
+     */
+    @Override
+    public String reportError(EObject object, String message) {
+        errorsOccurred = true;
+        acceptor.acceptError(message, object, null,
+                ValidationMessageAcceptor.INSIGNIFICANT_INDEX, null);
+        return message;
+    }
+
+    /**
+     * Report the given message as an error on the current object.
+     * 
+     * Unfortunately, there is no way to provide a path and a line number to the
+     * ValidationMessageAcceptor as messages can only be reported directly as
+     * EObjects. While it is not an ideal solution, this method composes a
+     * messages indicating the location of the error and reports this on the
+     * object currently under validation. This way, the error message is not
+     * lost, but it is not necessarily reported precisely at the location of the
+     * actual error.
+     */
+    @Override
+    public String reportError(Path file, Integer line, String message) {
+        errorsOccurred = true;
+        String fullMessage = "Error in " + file.toString() + " on line "
```

Comment:
I think the convention so far in other error reporters has been `ERROR: ...`. I think we should be consistent.

Reply:
I wouldn't say that this is a convention. This is only done in the `DefaultErrorReporter` which we don't really use anywhere. The error reported here will show up as an error marker in Epoch, or as a diagnostic in the output of lfc. I chose this text in an attempt to make it clearer that the location at which the marker is shown isn't precisely the location at which the error actually occurred. I am happy for suggestions on how to improve this, but I don't think using an `ERROR: ` prefix would be an improvement in this regard (The error marker already makes clear that this is an error).

Reply:
It's not just the `DefaultErrorReporter`. Again, since all our error reporters are scattered, it is easy to miss these things.
This convention is also used in [EclipseErrorReporter.java](https://github.com/lf-lang/lingua-franca/blob/570e218318415a408a121534903ce8e7dffba387/org.lflang/src/org/lflang/generator/EclipseErrorReporter.java#L98) and in our C target `warning_print` and `error_print`. I [found](https://github.com/lf-lang/lingua-franca/blob/570e218318415a408a121534903ce8e7dffba387/org.lflang/src/org/lflang/FileConfig.java#L828) [some](https://github.com/lf-lang/lingua-franca/blob/570e218318415a408a121534903ce8e7dffba387/org.lflang/src/org/lflang/generator/GeneratorBase.xtend#L601) [other](https://github.com/lf-lang/lingua-franca/blob/570e218318415a408a121534903ce8e7dffba387/org.lflang/src/org/lflang/generator/ts/TSGenerator.kt#L193) examples too. I was just pointing out that we should probably be consistent.

Reply:
@Soroosh129: `warning_print` and `error_print` are _runtime errors_. We currently don't have any measure of consistency across runtime implementations when it comes to error reporting. While we could formulate this as a goal, it has little to do with the issue at hand, which is how we format _compile errors_. I propose we create a separate issue for this, because we need to first reach consensus on what convention to use, and that seems outside of the scope of this PR.

Reply:
I think you misunderstood my comment. I'm not sure why you are focusing on `warning_print` and `error_print`.

Our two other descendants of the ErrorReporter interface (EclipseErrorReporter and DefaultErrorReporter)  use this `ERROR:` and `WARNING:` format.

Reply:
I'm focusing on `warning_print` and `error_print` because you mentioned them. I agree with @cmnrd that it's not super helpful to print `ERROR: ` in front of something that is already graphically marked as an error, but it is besides the point. This PR is about fixing an issue in our validator, not about achieving uniformity in error reporting. You rightly point out that there are inconsistencies, but I don't agree that there is a convention. Again, I propose to elect one. We can then implement it in a separate PR.

Reply:
Well, the point of my comment was that in some places, including the Eclipse IDE, some errors will read `ERROR: ...` and some will read `Error in ...`. I don't think we need to take the argument beyond that.

Reply:
I agree with you that this is an [issue](https://github.com/lf-lang/lingua-franca/issues/558), I just don't think that it should be considered as part of this PR.